test/vmtester/tests.cpp

```diff
@@ -168,3 +168,67 @@ TEST_F(evmc_vm_test, set_tracer)
     if (vm->set_tracer)
         vm->set_tracer(vm, tracer_callback, nullptr);
 }
+
+TEST_F(evmc_vm_test, precompile_test)
+{
+    if (!evmc_vm_has_capability(vm, EVMC_CAPABILITY_PRECOMPILES))
+        return;
+
+    // Iterate every address (as per EIP-1352)
+    for (size_t i = 0; i < 0xffff; i++)
+    {
+        evmc_address destination{{0,
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  static_cast<uint8_t>(i >> 8),
+                                  static_cast<uint8_t>(i & 0xff)}};
+        evmc_context* context = example_host_create_context();
+        evmc_message msg{
+            EVMC_CREATE,   0, 0, 65536, destination, evmc_address{}, NULL, 0, evmc_uint256be{},
+            evmc_bytes32{}};
+
+        evmc_result result = vm->execute(vm, context, EVMC_MAX_REVISION, &msg, nullptr, 0);
```

Comment:
Should even set `context` to `nullptr` according to the EIP.

Reply:
Yes.

Reply:
Actually it may be needed to check `callvalue`.

Reply:
Both are available in the message.

Reply:
Oh right, that's great.